src/items.rs

```diff
@@ -225,10 +227,7 @@ impl<'a> FmtVisitor<'a> {
         }
 
         // A conservative estimation, to goal is to be over all parens in generics
-        let args_start = generics.ty_params
-                                 .last()
-                                 .map(|tp| end_typaram(tp))
-                                 .unwrap_or(span.lo);
+        let args_start = generics.ty_params.last().map(|tp| end_typaram(tp)).unwrap_or(span.lo);
```

Comment:
Wouldn't an heuristic based on the characteristics of the chain (number on elements, length of elements, fields vs method call) be better ?
I personnally prefer that rustfmt doesn't take too much hints from the source, so that I can give it a very badly formatted file, and it doesn't take bad hints from my garbage formatting.